src/mgr/ActivePyModules.cc

```diff
@@ -880,6 +880,75 @@ PyObject* ActivePyModules::with_unlabled_perf_counters(
   return f.get();
 }
 
+PyObject* ActivePyModules::with_perf_counters(
+    std::function<void(PerfCounterInstance& counter_instance, PerfCounterType& counter_type, PyFormatter& f)> fct,
+    const std::string &svc_name,
+    const std::string &svc_id,
+    const std::string_view &counter_name,
+    const std::string_view &sub_counter_name,
+    const std::vector<std::pair<std::string_view,std::string_view>> &labels) const
+{
+  PyFormatter f;
+  /*
+    The resolved counter path, they are of the format
+    <counter_name>.<sub_counter_name> If the counter name has labels, then they
+    are segregated via NULL delimters.
+
+    Eg:
+      - labeled counter:
+        "osd_scrub_sh_repl^@level^@shallow^@pooltype^@replicated^@.successful_scrubs_elapsed"
+      - unlabeled counter: "osd.stat_bytes"
+  */
+  std::string resolved_path;
+  // NOTE: This should ideally be the resolved_path, but since the resolved path
+  // contains NULL delimiters and we cannot have an array section whose key has
+  // NULL characters, we will just use the counter_name to extract the value at
+  // the call site
+  f.open_array_section(counter_name);
+
+  // Construct the resolved path
+  if (!labels.empty()){
+    // NOTE: The 'ceph::perf_counters::key_create()' API requires a static
+    // array. Hence we are assuming that a performance counter will never have
+    // more than 100 labels.
+    ceph::perf_counters::label_pair labels_array[100];
```

Comment:
Please make it a constant that is clearly declared somewhere

Reply:
done! https://github.com/ceph/ceph/pull/61997/commits/8f8b7759ec0eeb75fcd5f585b66ae60e4e63425d#diff-50ab66411d9293d402a15e00ed6843a4d37889c616873e69534e609c210f72ecR914

Reply:
resolving this, because we no longer require this magic number